pkgs/http_profile/lib/http_profile.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+/// Records information about an HTTP request.
+final class HttpClientRequestProfile {
+  /// Determines whether HTTP profiling is enabled or not.
+  ///
+  /// The value can be changed programmatically or through the DevTools Network
+  /// UX.
+  static bool get profilingEnabled => HttpClient.enableTimelineLogging;
+  static set profilingEnabled(bool enabled) =>
+      HttpClient.enableTimelineLogging = enabled;
+
+  String? requestMethod;
```

Comment:
Can you expand the commit message (I usually edit the first PR comment then copy it to the message when I squash/merge) to include more details about how this class will be used by the SDK/DevTools?

What code will fill these fields in?

Reply:
Update the commit message.